L89-core-remove-grpc-insecure-channel-creation-api.md

```diff
@@ -0,0 +1,142 @@
+    terminates. See grpc_channel_args definition for more on this. */
+GRPCAPI grpc_channel* grpc_channel_create(const char* target,
+                                          grpc_channel_credentials* creds,
+                                          const grpc_channel_args* args);
+```
+
+Third, rename `grpc_server_add_secure_http2_port` to `grpc_server_add_http2_port` and move it from `grpc_security.h` to `grpc.h`.
+
+```
+/** Add a HTTP2 over an encrypted link over tcp listener.
+   Returns bound port number on success, 0 on failure.
+   REQUIRES: server not started */
+GRPCAPI int grpc_server_add_http2_port(grpc_server* server, const char* addr,
+                                       grpc_server_credentials* creds);
+```
+
+Fourth, remove `grpc_insecure_channel_create_from_fd` and `grpc_server_add_insecure_channel_from_fd`,
+and add `grpc_channel_create_from_fd` and `grpc_server_add_channel_from_fd` instead. Note that the new APIs only support
+either insecure channel credentials (at client-side) or insecure server credentials (at server-side), and using other types of
+credentials will result in a failure. In future, the APIs will be improved to support other types of credentials.
+
+
+```
+/** Create a secure channel to 'target' using file descriptor 'fd' and passed-in
+    credentials. The 'target' argument will be used to indicate the name for
+    this channel. Note that this API currently only supports insecure channel
+    credentials. Using other types of credentials will result in a failure. */
+GRPCAPI grpc_channel* grpc_channel_create_from_fd(
+    const char* target, int fd, grpc_channel_credentials* creds,
+    const grpc_channel_args* args);
+
+/** Add the connected secure communication channel based on file descriptor 'fd'
+   to the 'server' and server credentials 'creds'. The 'fd' must be an open file
+   descriptor corresponding to a connected socket. Events from the file
+   descriptor may come on any of the server completion queues (i.e completion
+   queues registered via the grpc_server_register_completion_queue API).
+   Note that this API currently only supports inseure server credentials
+   Using other types of credentials will result in a failure.
+   TODO(hork): add channel_args to this API to allow endpoints and transports
+   created in this function to participate in the resource quota feature. */
+GRPCAPI void grpc_server_add_channel_from_fd(grpc_server* server,
+                                             int fd,
+                                             grpc_server_credentials* creds);
+
+```
+Fifth, remove `grpc_insecure_channel_create` and `grpc_server_add_insecure_http2_port` from `grpc.h`. After this change takes effect, the only way of creating gRPC insecure channels will be first creating insecure channel or server credentials, and then invoking `grpc_channel_create` or `grpc_server_add_http2_port` at client or server side, respectively.
+
+## Rationale
```

Comment:
I'm curious if the issue https://github.com/grpc/grpc/issues/27733 (which I mentioned in https://github.com/grpc/grpc/pull/25586#issuecomment-944154028) will be resolved by the changes in the implementation of this proposal. If so, that's very positive and it should be mentioned here.

If not, perhaps we should mention it in the "Open issues" section, since in our own codebase, we're not always honoring the mutual exclusivity of  grpc and grpc_unsecure and we do have targets (e.g. some tests) that transitively depend on both `grpc` and `grpc_unsecure`.

Reply:
Ah, I do not think there is a need to keep `_nosec` end2end tests any more as we removed all code related to insecure channel creation (e.g., `grpc_insecure_channel_create`) API in this gRFC, that are exercised by `_nosec` tests.

Reply:
I think that's a good point. If we no longer need the `_nosec` tests after merging this PR (do other reviewers agree with this statement?), we should definitely get rid of those tests in a followup PR. There's actually quite a bit of complexity in how the _SEC vs _NOSEC end2end tests get configured and run, so this cleanup will be very welcome.

Reply:
Yes, I agree, we no longer need the `_nosec` tests.  I'm tempted to say that we should just remove those as part of Yihua's existing PR, but the PR is already quite large and invasive as it is, so I guess let's do that as a separate step.